src/hooks/index.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useEffect, useState } from 'react';
+
+export const useToggle = (initValue = false): [boolean, () => void] => {
+  const [flag, setFlag] = useState(initValue);
+
+  const toggle = useCallback(() => {
+    setFlag(!flag);
+  }, [flag]);
```

Comment:
This function will be re-created after every `flag` change.
This could get rid of any re-creation if you just replaced this with :
```suggestion
  const toggle = useCallback(() => {
    setFlag(flag => !flag);
  }, []);
```